communication.md

```diff
@@ -12,32 +12,43 @@ The SDK **MUST** use the [Telemetry ingest APIs](https://docs.newrelic.com/docs/
 
 ### User Agent
 
-All SDKs **MUST** send the following `User-Agent` header by default:
-`NewRelic-<language>-TelemetrySDK/<version>`
+The `User-Agent` header field is used to preform analytics on requests received by New Relic.
+In order to enable these analytics, all SDKs MUST include a `User-Agent` header in requests they make to New Relic.
+In addition to conforming to the specification defined in [RFC 7231](https://tools.ietf.org/html/rfc7231#section-5.5.3), the `User-Agent` header MUST include an SDK `product` identifier as its first entry.
 
-Additionally, SDKs **MUST** implement an API which allows appending product
-name and version to the `User-Agent` header in accordance with
-[RFC 7231](https://tools.ietf.org/html/rfc7231#section-5.5.3).
+```
+User-Agent  = sdk-id *( RWS ( product / comment ) )
+sdk-id      = sdk-name "/" sdk-version
+sdk-name    = "NewRelic-" language "-TelemetrySDK"
+sdk-version = token
+```
 
-The string appended to the `User-Agent` header **MUST** be in the format: `product/product_version`.
+The `language` portion of the `sdk-name` needs to be the programming language the SDK is written for and the `sdk-version` is the version of the SDK.
+The rest of this syntax ([`RWS`](https://tools.ietf.org/html/rfc7230#section-3.2.3), [`product`](https://tools.ietf.org/html/rfc7231#section-5.5.3), [`comment`](https://tools.ietf.org/html/rfc7230#section-3.2.6), and [`token`](https://tools.ietf.org/html/rfc7230#section-3.2.6)) all use the meanings defined in [RFC 7231](https://tools.ietf.org/html/rfc7231) and [RFC 7230](https://tools.ietf.org/html/rfc7230)
 
-Example:
+Understanding which exporter was used to export data is an important dimension to have analytics on as well.
+Exporters that use the SDK need to be able to append a `product` identifier of their own to the `User-Agent` header.
+Therefore, all SDKs MUST provide a method to extend the `User-Agent` header field-value.
+This method SHOULD accept the exporter determined `product` identifier as an argument.
+The exact form and the validity of this `product` identifier SHOULD be left to the exporter to determine.
```

Comment:
This brings up a reasonable question, as to what is a valid product identifier? For example, is a product identifier allowed to contain slashes?

Reply:
My goal was to updated the [exporter specs](https://github.com/newrelic/newrelic-exporter-specs) with specifics about what should be appended, including a reference to the the linked RFC's guidelines on the matter.

I'm interested to hear what you think we should include in that specification @a-feld. My goal was to ensure that it followed [RFC 7230](https://tools.ietf.org/html/rfc7230#section-3.2.6), but didn't have any thoughts on added requirements. We probably want some guidelines there stating that the name needs to reflect the code being instrumented, but not sure what else.

@a-feld I'll sync with you via slack about this so we can iterate faster.